force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -122,20 +122,19 @@ public with sharing class ClassesObjectsMethods {
      * @return A string containing the names of scientists that contain the letter 'g'.
      */
     public static String findTheScientist() {
-        // The list of scientists' names
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
-
-        // The variable to store the concatenated string
-
-        // Loop through the list of scientists' names
-
-            // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
-
-        // Remove the last comma from the result string
-
-        return null; // Replace null with the variable you used to store the result
+        List<String> namesWithG = new List<String>();
+        String scientistNamesG;
+        for (String individual : scientistNames){
+            if (individual.containsAny('Gg')) {
+                namesWithG.add(individual);
+            } else {
+                continue;
+            }
+        }
+        scientistNamesG = String.join(namesWithG, ',');
```

Comment:
You actually don't need to call `removeEnd`. String.join only puts a comma BETWEEN values!